src/System.Security.Principal.Windows/src/System/Security/Principal/NTAccount.cs

```diff
@@ -371,8 +371,6 @@ private static IdentityReferenceCollection TranslateToSids(IdentityReferenceColl
             finally
             {
                 LsaHandle.Dispose();
```

Comment:
And this can still call Dispose on the InvalidHandle if there was an exception thrown before the `LsaHandle` was initialized with actual handle.

Reply:
added check if handle is valid...if I understood well